### PR TITLE
Clean up several minor issues and add TODOs

### PR DESCRIPTION
**Pull request by EraYaN**
Add `Argument*Exception`s now use proper `nameof` operators.

Added exception messages to quite a few `Argument*Exception`s.

Fixed re throwing to be proper syntax (without variable name).

Added a ton of null checks. (This is only a start, there are about 500 places that need proper null handling)

Added some TODOs to log certain exceptions.

### CODE REVIEW

File: Emby.Drawing.Skia/StripCollageBuilder.cs [hunk 1]
```diff
@@ -20,6 +20,9 @@ public StripCollageBuilder(IApplicationPaths appPaths, IFileSystem fileSystem)
 
         public static SKEncodedImageFormat GetEncodedFormat(string outputPath)
         {
+            if (outputPath == null)
+                throw new ArgumentNullException(nameof(outputPath));
```
Comment (Bond-009) [inline Emby.Drawing.Skia/StripCollageBuilder.cs]: No braces?

File: Emby.Server.Implementations/LiveTv/LiveTvDtoService.cs [hunk 2]
```diff
@@ -331,10 +332,9 @@ private void FillImages(SeriesTimerInfoDto dto, string seriesName, string progra
                     {
                         try
                         {
-                            dto.ParentBackdropImageTags = new string[]
-                        {
-                                _imageProcessor.GetImageCacheTag(program, image)
-                        };
+                            dto.ParentBackdropImageTags = new string[] {
```
Comment (Bond-009) [inline Emby.Server.Implementations/LiveTv/LiveTvDtoService.cs]: Braces on a newline

File: Emby.Server.Implementations/Playlists/PlaylistManager.cs [hunk 3]
```diff
@@ -457,8 +457,8 @@ private string NormalizeItemPath(string playlistPath, string itemPath)
 
         private static String MakeRelativePath(string folderPath, string fileAbsolutePath)
         {
-            if (String.IsNullOrEmpty(folderPath)) throw new ArgumentNullException("folderPath");
-            if (String.IsNullOrEmpty(fileAbsolutePath)) throw new ArgumentNullException("filePath");
+            if (String.IsNullOrEmpty(folderPath)) throw new ArgumentException("Folder path was null or empty.",nameof(folderPath));
```
Comment (Bond-009) [inline Emby.Server.Implementations/Playlists/PlaylistManager.cs]: Also braces?

File: Emby.Server.Implementations/Security/EncryptionManager.cs [hunk 4]
```diff
@@ -14,7 +14,7 @@ public class EncryptionManager : IEncryptionManager
         /// <exception cref="System.ArgumentNullException">value</exception>
         public string EncryptString(string value)
         {
-            if (value == null) throw new ArgumentNullException("value");
+            if (value == null) throw new ArgumentNullException(nameof(value));
```
Comment (Bond-009) [inline Emby.Server.Implementations/Security/EncryptionManager.cs]: Same

File: Emby.Server.Implementations/Security/EncryptionManager.cs [hunk 5]
```diff
@@ -27,7 +27,7 @@ public string EncryptString(string value)
         /// <exception cref="System.ArgumentNullException">value</exception>
         public string DecryptString(string value)
         {
-            if (value == null) throw new ArgumentNullException("value");
+            if (value == null) throw new ArgumentNullException(nameof(value));
```
Comment (Bond-009) [inline Emby.Server.Implementations/Security/EncryptionManager.cs]: Same

File: Emby.Server.Implementations/Sorting/AiredEpisodeOrderComparer.cs [hunk 6]
```diff
@@ -16,6 +16,12 @@ class AiredEpisodeOrderComparer : IBaseItemComparer
         /// <returns>System.Int32.</returns>
         public int Compare(BaseItem x, BaseItem y)
         {
+            if (x == null)
```
Comment (Bond-009) [inline Emby.Server.Implementations/Sorting/AiredEpisodeOrderComparer.cs]: Same

**Comment by Bond-009** [comment]
Looks good, but I would add braces around the ones that don't have any.

**Comment by EraYaN** [comment]
I didn't want to reformat the document past the very basic functional fixes. That would make this an even harder to read pull request.

Should I still?

**Comment by EraYaN** [comment]
Alright one can look though all static methods to find ones that should not exist. Or really should be on another class.

Also the verbs are very often wrong. Tons of `Get*` methods should be `Convert*` or if move tot the proper class `As*`.

**Comment by joshuaboniface** [comment]
Currently failing Drone checks with this error:

```
/usr/share/dotnet/sdk/2.2.102/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.CrossTargeting.targets(31,5): error : The 'Publish' target is not supported without specifying a target framework. The current project targets multiple frameworks, please specify the framework for the published application. [/drone/src/ThirdParty/taglib-sharp/src/taglib-sharp.csproj] 
```

**Comment by EraYaN** [comment]
I know, the fix is in the build system commit. Didn’t want to put the same fix everywhere.

I can pull it in here as well, if needed.

**Comment by EraYaN** [comment]
Alright I added a merge commit made with `git merge -s recursive -X ours dev` so GitHub feels it's "clean."

File: RSSDP/DeviceEventArgs.cs [hunk 7]
```diff
@@ -18,14 +18,14 @@ public sealed class DeviceEventArgs : EventArgs
...
-        {
-            if (device == null) throw new ArgumentNullException("device");
+		/// <summary>
+		/// Constructs a new instance for the specified <see cref="SsdpDevice"/>.
+		/// </summary>
+		/// <param name="device">The <see cref="SsdpDevice"/> associated with the event this argument class is being used for.</param>
+		/// <exception cref="System.ArgumentNullException">Thrown if the <paramref name="device"/> argument is null.</exception>
+		public DeviceEventArgs(SsdpDevice device)
+		{
+			if (device == null) throw new ArgumentNullException(nameof(device));
```
Comment (Bond-009) [inline RSSDP/DeviceEventArgs.cs]: What's up with this indentation.

File: RSSDP/DeviceUnavailableEventArgs.cs [hunk 8]
```diff
@@ -5,55 +5,55 @@
...
-        /// </summary>
-        public DiscoveredSsdpDevice DiscoveredDevice
-        {
-            get { return _DiscoveredDevice; }
-        }
-
-        #endregion
-    }
-}
+	/// <summary>
```
Comment (Bond-009) [inline RSSDP/DeviceUnavailableEventArgs.cs]: Indentation?

File: RSSDP/HttpRequestParser.cs [hunk 9]
```diff
@@ -8,70 +8,70 @@
...
-            var parts = data.Split(' ');
-            if (parts.Length < 2) throw new ArgumentException("Status line is invalid. Insufficient status parts.", "data");
-
-            message.Method = new HttpMethod(parts[0].Trim());
-            Uri requestUri;
-            if (Uri.TryCreate(parts[1].Trim(), UriKind.RelativeOrAbsolute, out requestUri))
-                message.RequestUri = requestUri;
-            else
-                System.Diagnostics.Debug.WriteLine(parts[1]);
+	/// <summary>
```
Comment (Bond-009) [inline RSSDP/HttpRequestParser.cs]: Indentation?

File: RSSDP/HttpResponseParser.cs [hunk 10]
```diff
@@ -8,82 +8,82 @@
...
-            if (parts.Length < 2) throw new ArgumentException("data status line is invalid. Insufficient status parts.", "data");
-
-            message.Version = ParseHttpVersion(parts[0].Trim());
-
-            int statusCode = -1;
-            if (!Int32.TryParse(parts[1].Trim(), out statusCode))
-                throw new ArgumentException("data status line is invalid. Status code is not a valid integer.", "data");
-
-            message.StatusCode = (HttpStatusCode)statusCode;
+	/// <summary>
```
Comment (Bond-009) [inline RSSDP/HttpResponseParser.cs]: Indentation?

File: RSSDP/IEnumerableExtensions.cs [hunk 11]
```diff
@@ -5,12 +5,12 @@
 
 namespace Rssdp.Infrastructure
 {
-    internal static class IEnumerableExtensions
-    {
-        public static IEnumerable<T> SelectManyRecursive<T>(this IEnumerable<T> source, Func<T, IEnumerable<T>> selector)
-        {
-            if (source == null) throw new ArgumentNullException("source");
-            if (selector == null) throw new ArgumentNullException("selector");
+	internal static class IEnumerableExtensions
```
Comment (Bond-009) [inline RSSDP/IEnumerableExtensions.cs]: Indentation?

File: DvdLib/DvdLib.csproj [hunk 12]
```diff
@@ -1,5 +1,9 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
+  <ItemGroup>
+    <Compile Include="..\SharedVersion.cs" Link="SharedVersion.cs" />
```
[the remark this answers is not in the document]
Comment (EraYaN) [reply DvdLib/DvdLib.csproj]: I think so. There was no proper version in this project as at. So this is better.

[the remark this answers is not in the document]
Comment (Bond-009) [reply DvdLib/DvdLib.csproj]: Ideally it should be a standalone project

**Comment by EraYaN** [comment]
When this get's merged, the drone build breaks. So I either pull in changes from the build PR to unbreak the drone builds or we merge that one this weekend as well.

File: Emby.Server.Implementations/ScheduledTasks/ScheduledTaskWorker.cs [hunk 13]
```diff
@@ -29,7 +29,6 @@ public class ScheduledTaskWorker : IScheduledTaskWorker
         /// <value>The scheduled task.</value>
         public IScheduledTask ScheduledTask { get; private set; }
 
-        /// <summary>
```
Comment (Bond-009) [inline Emby.Server.Implementations/ScheduledTasks/ScheduledTaskWorker.cs]: Removed opening tag?

Comment (EraYaN) [reply Emby.Server.Implementations/ScheduledTasks/ScheduledTaskWorker.cs]: Not by hand.

**Comment by EraYaN** [comment]
The braces can be added in a big reformat/refactor PR. If this can be merged this weekend I can do it tomorrow.

**Pull request merged**